ecs/src/nox/ecs/component/Transform.cpp

```diff
@@ -7,6 +7,19 @@
 #include <glm/gtc/matrix_transform.hpp>
 #include <nox/util/json_utils.h>
 
+nox::ecs::Transform(const glm::vec2& position, 
```

Comment:
We are not allowing constructors with arguments, other than:
And unless you are doing calculations within the constructor it should not be overloaded. Just do regular default initialization in the header file with {}.
```
Component(const EntityId& entityId, EntityManager* entityManager);